OpenSim/Simulation/Manager/Manager.cpp

```diff
@@ -624,22 +625,22 @@ hasStateStorage() const
 // INTEGRATION
 //-----------------------------------------------------------------------------
 
-bool Manager::
-integrate(SimTK::State& s, double finalTime)
+SimTK::State Manager::
+integrate(double finalTime)
 {
     int step = 0; // for AnalysisSet::step()
 
-    if (!_integ) {
-        throw Exception("Manager::integrate(): "
-            "Integrator has not been set. Construct the Manager "
-            "with an integrator, or call Manager::setIntegrator().");
+    if (_timeStepper == nullptr) {
+        throw Exception("Manager::integrate(): Manager has not be "
+            "initialized. Call Manager::initialize() first.");
     }
 
     // CLEAR ANY INTERRUPT
     // Halts must arrive during an integration.
     clearHalt();
 
     // CHECK SPECIFIED DT STEPPING
+    SimTK::State s = _integ->getState();
```

Comment:
@carmichaelong this is the line causing the issue. Here we get a local copy that we then realize to velocity, but the internal state of the integrator is unchanged. If I update this to be  `const SimTK::State&` the issue goes away and we do not need the `model.realizeVelocity()` call I added. There is another issue that we didn't get an error that the state was not adequately realized. If I invoke `myStation->getLocationInGround(initState)` we do get at the cache is does not satisfy its prerequisites.

Reply:
I also updated `Manager`:
```cpp
    const SimTK::State& integrate(double finalTime);
    const SimTK::State& getState() const;
```
Now, return const ref. This did require updating Analysi:begin() and end() to also take const SimTK::State&`,  but I believe we have now eliminated unnecessary local copies of the `State`.

I'll push these changes once I see all the tests pass locally.